diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..d66d08b
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,50 @@
+#!/usr/bin/make -f
+#
+#	Copyright (C) 2007-2008 Jari Aalto
+#
+#	This program is free software; you can redistribute it and/or
+#	modify it under the terms of the GNU General Public License as
+#	published by the Free Software Foundation; either version 2 of the
+#	License, or (at your option) any later version
+#
+#	This program is distributed in the hope that it will be useful, but
+#	WITHOUT ANY WARRANTY; without even the implied warranty of
+#	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#	General Public License for more details.
+#
+#	Visit <http://www.gnu.org/copyleft/gpl.html>
+
+MAKEFILE  = Makefile.unx
+
+DESTDIR		=
+prefix		= /usr
+exec_prefix	= $(prefix)
+man_prefix	= $(prefix)/share
+
+INSTALL		= /usr/bin/install
+INSTALL_BIN	= $(INSTALL) -m 755
+INSTALL_DATA	= $(INSTALL) -m 644
+INSTALL_OBJS	= pngcheck
+
+MANDIR		= $(DESTDIR)$(man_prefix)/man/man1
+BINDIR		= $(DESTDIR)$(exec_prefix)/bin
+
+all:
+	$(MAKE) -f $(MAKEFILE) LIBS="-lz"
+
+distclean: clean
+
+clean:
+	$(MAKE) -f $(MAKEFILE) clean
+
+install-bin:
+	$(INSTALL_BIN) -d $(BINDIR)
+	$(INSTALL_BIN) -s $(INSTALL_OBJS) $(BINDIR)
+
+install-man:
+	$(INSTALL_BIN) -d $(MANDIR)
+	$(INSTALL_DATA) *.1 $(MANDIR)
+
+install: install-bin
+
+# End of file
